### PR TITLE
Remove Short Circuit read/write

### PR DESCRIPTION
**Pull request by apc999**
### What changes are proposed in this pull request?

Remove source code related to Short Circuit read/write

### Why are the changes needed?

- In K8s env (increasingly more popular), support short-ciruit I/O is challenging
- We shall use SDK instead

### Does this PR introduce any user facing changes?

No more Short Circuit read/write

### CODE REVIEW

**Comment by ssz1997** [comment]
Shall the short-circuit-related metrics be removed in this PR as well? Or metrics cleanup will be an another process?

**Comment by apc999** [comment]
> Shall the short-circuit-related metrics be removed in this PR as well? Or metrics cleanup will be an another process?

I will make  a separate PR to clean up short-circuit related metrics

**Review by jiacheliu3 — COMMENTED** [review]
Some hard coded tests are complaining:
```
2023-07-14T07:29:09.9167401Z 07:29:09.915 [ERROR] Failures: 
2023-07-14T07:29:09.9168088Z 07:29:09.915 [ERROR] alluxio.cli.fsadmin.report.MetricsCommandTest.metrics
2023-07-14T07:29:09.9169006Z 07:29:09.915 [ERROR]   Run 1: MetricsCommandTest.metrics:62->checkIfOutputValid:145 
2023-07-14T07:29:09.9175392Z Expected: iterable containing ["Cluster.BytesReadDomain  (Type: GAUGE, Value: 4145.73KB)", "Cluster.BytesReadDomainThroughput  (Type: GAUGE, Value: 29.97MB/MIN)", "Cluster.BytesReadLocal  (Type: GAUGE, Value: 11.47GB)", "Cluster.BytesReadLocalThroughput  (Type: GAUGE, Value: 117.42MB/MIN)", "Cluster.BytesReadRemote  (Type: GAUGE, Value: 401.79MB)", "Cluster.BytesReadRemoteThroughput  (Type: GAUGE, Value: 518.36MB/MIN)", "Cluster.BytesReadUfsAll  (Type: GAUGE, Value: 509.47MB)", "Cluster.BytesReadUfsThroughput  (Type: GAUGE, Value: 728.16KB/MIN)", "Cluster.BytesWrittenDomain  (Type: GAUGE, Value: 62.43MB)", "Cluster.BytesWrittenDomainThroughput  (Type: GAUGE, Value: 1202.37KB/MIN)", "Cluster.BytesWrittenRemote  (Type: GAUGE, Value: 22.98KB)", "Cluster.BytesWrittenRemoteThroughput  (Type: GAUGE, Value: 8.03MB/MIN)", "Cluster.BytesWrittenUfsAll  (Type: GAUGE, Value: 317.70KB)", "Cluster.BytesWrittenUfsThroughput  (Type: GAUGE, Value: 33.46KB/MIN)", "Cluster.CapacityTotal  (Type: GAUGE, Value: 1,154,531,246,129,122)", "Master.CompleteFileOps  (Type: COUNTER, Value: 813)", "Master.UfsSessionCount-Ufs:_alluxio_underFSStorage  (Type: COUNTER, Value: 8,535)", "Master.UfsSessionCount-Ufs:file:___Users_alluxio_alluxioMountedFolder  (Type: COUNTER, Value: 1,231)", "Master.getMetrics.User:alluxio  (Type: TIMER, Value: 4)"]
2023-07-14T07:29:09.9178398Z      but: item 2: was "Cluster.BytesReadRemote  (Type: GAUGE, Value: 401.79MB)"
2023-07-14T07:29:09.9179129Z 07:29:09.915 [ERROR]   Run 2: MetricsCommandTest.metrics:62->checkIfOutputValid:145 
2023-07-14T07:29:09.9184592Z Expected: iterable containing ["Cluster.BytesReadDomain  (Type: GAUGE, Value: 4145.73KB)", "Cluster.BytesReadDomainThroughput  (Type: GAUGE, Value: 29.97MB/MIN)", "Cluster.BytesReadLocal  (Type: GAUGE, Value: 11.47GB)", "Cluster.BytesReadLocalThroughput  (Type: GAUGE, Value: 117.42MB/MIN)", "Cluster.BytesReadRemote  (Type: GAUGE, Value: 401.79MB)", "Cluster.BytesReadRemoteThroughput  (Type: GAUGE, Value: 518.36MB/MIN)", "Cluster.BytesReadUfsAll  (Type: GAUGE, Value: 509.47MB)", "Cluster.BytesReadUfsThroughput  (Type: GAUGE, Value: 728.16KB/MIN)", "Cluster.BytesWrittenDomain  (Type: GAUGE, Value: 62.43MB)", "Cluster.BytesWrittenDomainThroughput  (Type: GAUGE, Value: 1202.37KB/MIN)", "Cluster.BytesWrittenRemote  (Type: GAUGE, Value: 22.98KB)", "Cluster.BytesWrittenRemoteThroughput  (Type: GAUGE, Value: 8.03MB/MIN)", "Cluster.BytesWrittenUfsAll  (Type: GAUGE, Value: 317.70KB)", "Cluster.BytesWrittenUfsThroughput  (Type: GAUGE, Value: 33.46KB/MIN)", "Cluster.CapacityTotal  (Type: GAUGE, Value: 1,154,531,246,129,122)", "Master.CompleteFileOps  (Type: COUNTER, Value: 813)", "Master.UfsSessionCount-Ufs:_alluxio_underFSStorage  (Type: COUNTER, Value: 8,535)", "Master.UfsSessionCount-Ufs:file:___Users_alluxio_alluxioMountedFolder  (Type: COUNTER, Value: 1,231)", "Master.getMetrics.User:alluxio  (Type: TIMER, Value: 4)"]
2023-07-14T07:29:09.9187204Z      but: item 2: was "Cluster.BytesReadRemote  (Type: GAUGE, Value: 401.79MB)"
```

Other than that, I browsed all changes and I think they mostly LGTM. I need to take a closer look tonight. From the scale of this PR (~2000 lines), I figure I can do a full scan of all changed lines.

If a change grows much larger than that (>3000 lines or even much larger than that), I believe it's close to impossible to do a thorough review. And for those PRs I'd recommend:
1. Break it into smaller reviewable parts and allow leftovers in between, as long as leftovers do not interfere other code;
2. Or try to allow ONLY removing code files as a whole - try to do changes as sparingly as possible because the changes cannot be reviewed or noticed at all. If a change in-line is compulsory, segregate that to one commit and log it in description. That was the approach we collectively took in https://github.com/Alluxio/alluxio/pull/17638 with ~40k lines of change.

File: dora/core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java [hunk 1]
```diff
@@ -115,49 +114,27 @@ public static BlockInStream create(FileSystemContext context, BlockInfo info,
     }
 
     AlluxioConfiguration alluxioConf = context.getClusterConf();
-    boolean shortCircuit = alluxioConf.getBoolean(PropertyKey.USER_SHORT_CIRCUIT_ENABLED);
```
Comment (jiacheliu3) [inline dora/core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java:118]: In the comment of this method, there's "short-circuit" mentioned.

File: dora/core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java [hunk 2]
```diff
@@ -57,9 +57,6 @@ private Factory() {} // prevent instantiation
     public static DataWriter create(FileSystemContext context, long blockId, long blockSize,
         WorkerNetAddress address, OutStreamOptions options) throws IOException {
       AlluxioConfiguration alluxioConf = context.getClusterConf();
-      boolean shortCircuit = alluxioConf.getBoolean(PropertyKey.USER_SHORT_CIRCUIT_ENABLED);
```
Comment (jiacheliu3) [inline dora/core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java:60]: In `GrpcDataWriter` there are comments mentioning "short-circuit". I found those occurrences by searching "shortcircuit" and "short-circuit" and "short circuit".

Comment (jiacheliu3) [reply dora/core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java:60]: Some other occurrences include `Format.java`, `package-info.java`, `BlockInStream.java`, `DecommissionWorkerCommand` etc.

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Thanks for the work!

Found some extra occurrences of the term short circuit, which are not included here. It depends on your intention of this PR whether we want to fix those individual occurrences scattered every where, or you may create a separate one to address those comment-only occurrences.

Some other relevant changes you won't be able to cover in this PR, but might need extra awareness:
1. In web UI there are at least some gauges for short circuit related metrics for display
2. Some relevant sections in the doc pages

File: dora/core/common/src/main/java/alluxio/wire/MasterWebUIMetrics.java [hunk 3]
```diff
@@ -70,21 +65,12 @@ public MasterWebUIMetrics() {
   }
 
   /**
-   * Gets cache hit local.
-   *
-   * @return the cache hit local
-   */
-  public String getCacheHitLocal() {
```
[the remark this answers is not in the document]
Comment (apc999) [reply dora/core/common/src/main/java/alluxio/wire/MasterWebUIMetrics.java:77]: agree. let's revisit the metrics name later. I will leave a TODO here

File: dora/core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java (hunk 2 above)
Comment (apc999) [reply dora/core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java:60]: all removed

**Review by apc999 — COMMENTED** [review]
Java comment part addressed. PTAL

For webui/metrics, we shall have a separate round of revamp

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the work!

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**